locator/abstract_replication_strategy.cc

```diff
@@ -479,12 +479,12 @@ host_id_vector_replica_set vnode_effective_replication_map::do_get_replicas(cons
     return it->second;
 }
 
-host_id_vector_replica_set vnode_effective_replication_map::get_replicas(const token& tok) const {
-    return do_get_replicas(tok, false);
+host_id_vector_replica_set vnode_effective_replication_map::get_replicas(const token& tok, bool is_vnode) const {
+    return do_get_replicas(tok, is_vnode);
 }
```

Comment:
What does is_vnode mean? That it's one of some node's tokens?

Ok, that's what the patch changelog says.

But why is it important? All the data paths will look for the partition key's token. I feel it clutters the interface and slows down the data path (by a tiny bit) for an unimportant bit in the control path.

Reply:
btw, such distinctions are better in the method name. Parameters are good when a caller can call with many values, but when each call site calls with a constant, it means you're multiplexing different things in a single function.

Reply:
> What does is_vnode mean? That it's one of some node's tokens?

Yes.  From the vnode perspective it's the vnode's first token (inclusive)

> 
> Ok, that's what the patch changelog says.
> 
> But why is it important? All the data paths will look for the partition key's token. I feel it clutters the interface and slows down the data path (by a tiny bit) for an unimportant bit in the control path.

The replication map is indexed by the vnode tokens.
When an arbitrary token is looked up we revert to a binary search (std::lower_bound) in `sorted_tokens`.
Although the latter has O(log N) complexity which isn't too bad it doing random access in memory and trashes the dcache, and it is done on every range where in a large cluster we can have tens of thousands of those.

Reply:
> btw, such distinctions are better in the method name. Parameters are good when a caller can call with many values, but when each call site calls with a constant, it means you're multiplexing different things in a single function.

The function (`vnode_effective_replication_map::do_get_replicas`) is already multiplxed:
```
   const token& key_token = _rs->natural_endpoints_depend_on_token()
        ? (is_vnode ? tok : _tmptr->first_token(tok))
        : default_replication_map_key;
```

Another approach I contemplated about was to define `struct search_token` that would have a boolean class indicating `is_vnode` (maybe using a better term) that would be passed down to `do_get_replicas`.

Reply:
Ok, if it's already there, of course we can use it, but the criticisms for the API apply.